server/server_test.go

```diff
@@ -365,13 +378,15 @@ func TestCompileV1UnsafeBuiltin(t *testing.T) {
 		t.Fatalf("Expected bad request but got %v", f.recorder)
 	}
 
-	expected := `{
+	actual := []byte(f.recorder.Body.String())
+
+	expected := []byte(`{
   "code": "invalid_parameter",
   "message": "unsafe built-in function calls in query: http.send"
-}`
+}`)
 
-	if f.recorder.Body.String() != expected {
-		t.Fatalf(`Expected %v but got: %v`, expected, f.recorder.Body.String())
+	if !JSONBytesEqual(expected, actual) {
```

Comment:
Nit: we have a helper for asserting on the response already. See other cases in this file. Something like:

```
err := f.v1(http.MethodPost, "/compile", `{"query": ...}`, http.StatusBadRequest, `{"code": ..., "message": ...}`)
if err != nil {
  t.Fatal(err)
}
```

`f.v1` will return an error if the response status or body does not match the expected values. It'll do the right thing and compare value equality, not string equality.

Similar to the templating comment above, this is outside the scope of your changes. If you can do the refactoring that would be awesome, but if not, I'm happy to do a follow-up PR. Let me know.

Reply:
I got these in.